engine/class_modules/sc_death_knight.cpp

```diff
@@ -3957,16 +3978,11 @@ struct helchains_buff_t : public buff_t
     } );
   }
 
-  // Helchains ticks twice on buff application and buff expiration, hence the following overrides
-  // https://github.com/SimCMinMax/WoW-BugTracker/issues/390
-  void execute( int stacks, double value, timespan_t duration ) override
-  {
-    buff_t::execute( stacks, value, duration );
-    damage -> execute();
-  }
-
+  // Unholy pact ticks twice on buff expiration, hence the following override
+  // https://github.com/SimCMinMax/WoW-BugTracker/issues/675
   void expire_override( int expiration_stacks, timespan_t remaining_duration ) override
```

Comment:
The last two hits aren't affected by the buff, but buff_t::expire_override is called during buff_t::expire() while the buff is still technically active. You can circumvent that by calling action_t::schedule_execute() instead of action_t::execute(), this will create damage events that will fire after a 0s delay, but after the current operations are done.